src/formatDate.js

```diff
@@ -8,7 +8,65 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const YEAR_SHORT_FORMAT = 'YY';
+  const YEAR_LONG_FORMAT = 'YYYY';
+  const MONTH_FORMAT = 'MM';
+  const DAY_FORMAT = 'DD';
+  const CENTURY_21 = '20';
+  const CENTURY_20 = '19';
+  const MAX_YEAR = 30;
+  const DELIMITER = fromFormat.pop();
+  const SEPARATOR = toFormat.pop();
+  const dateParts = date.split(DELIMITER);
+  const shortYearIndex = fromFormat.indexOf(YEAR_SHORT_FORMAT);
+  const longYearIndex = fromFormat.indexOf(YEAR_LONG_FORMAT);
+  const formattedDateParts = [];
+
+  for (let i = 0; i < toFormat.length; i++) {
+    let index;
+    let year;
+
+    switch (toFormat[i]) {
+      case YEAR_SHORT_FORMAT:
+        index = shortYearIndex < 0 ? longYearIndex : shortYearIndex;
+        year = dateParts[index];
+
+        if (year.length === 4) {
+          formattedDateParts.push(year.slice(-2));
+        } else {
+          formattedDateParts.push(year);
+        }
+        break;
+
+      case MONTH_FORMAT:
+        index = fromFormat.indexOf(MONTH_FORMAT);
+        formattedDateParts.push(dateParts[index]);
+        break;
+
+      case DAY_FORMAT:
+        index = fromFormat.indexOf(DAY_FORMAT);
+        formattedDateParts.push(dateParts[index]);
+        break;
+
+      case YEAR_LONG_FORMAT:
+        index = longYearIndex < 0 ? shortYearIndex : longYearIndex;
+        year = dateParts[index];
+
+        if (year.length === 2) {
+          const century = year < MAX_YEAR ? CENTURY_21 : CENTURY_20;
+
+          formattedDateParts.push(century + year);
```

Comment:
The comparison 'year < MAX_YEAR' works because JavaScript coerces the string to a number in this context. However, it would be clearer to explicitly convert the string to a number before comparing it to MAX_YEAR for better readability and to avoid confusion.